app/controllers/sessions_controller.rb

```diff
@@ -0,0 +1,31 @@
+class SessionsController < ApplicationController
+  skip_before_action :ensure_logged_in
+
+  def new
+  end
+
+  def create
+    user = User.find_by(username_param)
+    if user && user.authenticate(password_param)
+      session[:user_id] = user.id
+      redirect_to root_path
+    else
+      render 'new'
+    end
+  end
+
+  def destroy
+    session[:user_id] = nil
+    redirect_to root_path
+  end
+
+  private
+    def username_param
+      params.require(:session).permit(:username)
```

Comment:
You don't need strong params here - they are only necessary when you are doing mass assignment to a model.